src/v2/BaseLoginRouter.ts

```diff
@@ -272,27 +288,32 @@ export default Router.extend({
 
     // Re-render the widget
     this.render(this.controller.constructor);
-  },
+  }
 
-  start: function() {
+  start() {
     const pushState = BrowserFeatures.supportsPushState();
     Router.prototype.start.call(this, { pushState: pushState });
-  },
+  }
 
-  hide: function() {
+  hide() {
     this.header.$el.hide();
-  },
+  }
 
-  show: function() {
+  show() {
     this.header.$el.show();
-  },
+  }
 
-  remove: function() {
+  remove() {
     this.unload();
     this.header.$el.remove();
     this.stopListening(this.appState);
     this.stopListening(this.settings);
     Bundles.remove();
     Backbone.history.stop();
-  },
-});
+  }
+}
+
+// Add "Events" to prototype for compatibility with existing code
+BaseLoginRouter.prototype.Events = Backbone.Events;
```

Comment:
This is for compatibility with existing code which is calling BaseLoginRouter.prototype.Events.  Class members (fields) aer not put on the prototype automatically in TS classes. (although this can be done with a babel transform which converts to CJS). We are planning to expose widget src as ESM so we need it to be compatible without transpile to CJS

Reply:
I think there probably can be some refactor/simplification around the existing code, but this PR is already so large. I am trying to reduce any chance of breaking something. So I'm mostly only changing things that are necessary for TS type exports. A few files I converted to TS/classes to verify that the courage types are working as expected (ran into issues converting some Views to classes, so rolled that back). There will definitely be follow up PRs to continue TS conversion and refine types, I think we can tackle this Router events thing in one of those PRs.